standard/statements.md

```diff
@@ -1112,33 +1112,50 @@ If the *foreach_statement* contains both or neither `ref` and `readonly`, the it
-- Otherwise, check for an enumerable interface:
-  - If among all the types `Tᵢ` for which there is an implicit conversion from `X` to `IEnumerable<Tᵢ>`, there is a unique type `T` such that `T` is not `dynamic` and for all the other `Tᵢ` there is an implicit conversion from `IEnumerable<T>` to `IEnumerable<Tᵢ>`, then the collection type is the interface `IEnumerable<T>`, the enumerator type is the interface `IEnumerator<T>`, and the iteration type is `T`.
-  - Otherwise, if there is more than one such type `T`, then an error is produced and no further steps are taken.
-  - Otherwise, if there is an implicit conversion from `X` to the `System.Collections.IEnumerable` interface, then the collection type is this interface, the enumerator type is the interface `System.Collections.IEnumerator`, and the iteration type is `object`.
-  - Otherwise, an error is produced, and no further steps are taken.
+- «GetEnumerator» is a `GetEnumerator` method.
+- «MoveNext» is a `MoveNext` method with a `bool` return type.
+- «IEnumerable»\<T> is the `System.Collections.Generic.IEnumerable<T>` interface.
+- «IEnumerator»\<T> is the `System.Collections.Generic.IEnumerator<T>` interface.
+
+In addition, the following modifications are made to the steps in §13.9.5.1:
+
+Before the process described in §13.9.5.1, the following steps are taken:
 
-The above steps, if successful, unambiguously produce a collection type `C`, enumerator type `E` and iteration type `T`, `ref T`, or `ref readonly T`. A `foreach` statement of the form
+- If the type `X` of *expression* is an array type then there is an implicit reference conversion from `X` to the `IEnumerable` interface (since `System.Array` implements this interface). The collection type is the `IEnumerable` interface, the enumerator type is the `IEnumerator` interface and the iteration type is the element type of the array type `X`. (sync)
+- If the type `X` of *expression* is `dynamic` then there is an implicit conversion from *expression* to the `IEnumerable` interface ([§10.2.10](conversions.md#10210-implicit-dynamic-conversions)). The collection type is the `IEnumerable` interface and the enumerator type is the `IEnumerator` interface. If the `var` identifier is given as the *local_variable_type* then the iteration type is `dynamic`, otherwise it is `object`. (sync only)
+
+If the process in §13.9.5.1 completes without produce a single collection type, enumerator type, and iteration type, the following steps are taken:
+
+- Otherwise, if there is an implicit conversion from `X` to the `System.Collections.IEnumerable` interface, then the collection type is this interface, the enumerator type is the interface `System.Collections.IEnumerator`, and the iteration type is `object`. (sync only)
```

Comment:
```suggestion
- If there is an implicit conversion from `X` to the `System.Collections.IEnumerable` interface, then the collection type is this interface, the enumerator type is the interface `System.Collections.IEnumerator`, and the iteration type is `object`.
```

Reply:
("Otherwise" at the start of a list is weird, and this is already in a sync-only section.)